gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -511,6 +512,36 @@ public GsonBuilder registerTypeAdapter(Type type, Object typeAdapter) {
     return this;
   }
 
+  /**
+   * Configures Gson for custom serialization and deserialization with Fill-in. Takes the type
+   * adapter, and wraps the result from the {@link TypeAdapter} into a {@link ReflectiveTypeAdapterFactory.Adapter}. Like
+   * {@link #registerTypeAdapter(Type, Object)}, it only applies to the type specified by
+   * the {@code type} parameter.
+   *
+   * @param baseType the type definition for the type adapter being registered
+   * @param objectAdapter This object must implement the {@link TypeAdapter} class
+   * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   */
+  public GsonBuilder registerTypeAdapterWithFillIn(final Type baseType, Object objectAdapter) {
```

Comment:
Instead of this method, why not create a TypeAdapter implementation that users can extend to get fill-in features. 
e.g., 
ReflectiveTypeAdapter that internally will use ReflectiveTypeAdapterFactory